ruby/tests/gc_test.rb

```diff
@@ -95,9 +95,15 @@ def test_generated_msg
     data = A::B::C::TestMessage.encode(from)
     to = A::B::C::TestMessage.decode(data)
 
-    from = get_msg_proto2
-    data = A::B::Proto2::TestMessage.encode(from)
-    to = A::B::Proto2::TestMessage.decode(data)
+    # This doesn't work for proto2 on JRuby because there is a nested required message.
```

Comment:
Does this mean that the plain Ruby implementation isn't validating required fields during parsing?

Reply:
Yeah, I didn't dig into the c-ruby side too much on it but I think it tracked down to it not checking things in the same order. There is a required field that is the same type as the message that contains it and, iirc, it turns into an infinite loop.

Reply:
That makes sense, it's just that the fact that this test was passing for C Ruby makes me think it is not validating required fields, which is a separate problem that we should probably fix at some point. This is a very weird test message because it's impossible to create a valid instance of it, given the recursive required field.